### ISSUE TITLE
[Feature Request]: Document deep link protocol length limits

### ISSUE THREAD

**Issue by theogravity** [electron/electron#40776]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

I'm trying to enable our electron app for Windows (working in MacOS), and after a day and half of troubleshooting, I discovered that there are character limits for deep links under Windows.

The limit isn't with the browser, but Windows OS `ShellExecute` having a limit of 2083 chars (in my personal testing, it was 2048).

Related Chromium ticket:

https://bugs.chromium.org/p/chromium/issues/detail?id=727909

```
In Chrome, if you attempt to an Application Protocol URL over 2046 characters, you’ll see the regular Security Prompt, but clicking its “Launch Application” button does nothing.
```

tl;dr: Custom protocols in Windows have a limit of 2048 - 2083 characters. If you have such a URL coming from Edge / Chrome, you'll get the prompt to open the app associated with the protocol, but on confirmation, nothing will get launched.

If you trim the URL length to less than 2048, your electron app will launch.

(You may be asking what are you putting in your custom protocol - for us it was a signed JWT, which can easily exceed that limit. We'll have to implement a device auth grant instead for our electron app.) 

Keywords: deep linking / deep link / windows / `setAsDefaultProtocolClient`

### Proposed Solution

Add documentation to deep links on the character limits for a URL when using a custom protocol by OS.

### Alternatives Considered

N/A.

**Comment by gabrielyotoo** [comment]
I am facing this issue too. Have you find any workarounds?
I noticed that it worked fine in Windows using Firefox browser, but failed with Chrome and Edge
